pkg/encryption/encryption.go

```diff
@@ -60,6 +65,22 @@ func (nonce *AESGCMNonce) Bytes() *[AESGCMNonceSize]byte {
 	return (*[AESGCMNonceSize]byte)(nonce)
 }
 
+// EncryptedPrivateKey is a private key that has been encrypted
+type EncryptedPrivateKey []byte
+
+// Decrypt decrypts the current key with the given key and nonce
+func (k EncryptedPrivateKey) Decrypt(cipher Cipher, key *Key, nonce *Nonce) (*Key, error) {
```

Comment:
Doesn't it make more sense to have the method on Cipher rather than the Key?

Reply:
Does it? My rationale is that the Key can be encrypted, hence it has its own method for this.

The Cipher also have Encrypt and Decrypt methods, but they are more general, to encrypt/decrypt any data. Does it make more sense to have `Cipher.EncryptKey` instead of `Key.Encrypt`, and `Cipher.DecryptEncryptedPrivateKey`  instead of `EncryptedPrivateKey.Decrypt`?

Reply:
`cipher.EncryptKey`, `cipher.DecryptKey`?

Reply:
Should we have `cipher.EncryptPath` and `cipher.DecryptPath` the same way with regards to path encryption instead of `Path.Encrypt` and `Path.Decrypt`?

Reply:
I think it's more a semantic question. I.e. `cipher decrypts/encrypts things` so to me it makes to have a method `EncryptSomething`... where as `PrivateKey` and `EncryptedPrivateKey` is mainly data rather than being the actor.

Reply:
So same should apply to Path, Metadata, whatever needs to be encrypted?! This Cipher type would grow huge of methods.

Reply:
Ok, you've convinced me.